File: Code/Iterator.dotNet/Iterator.dotNet/Iterator.dotNet/DataStructure/TreeNode.cs
using Iterator.Model;

namespace Iterator.DataStructure
{
    /// <summary>
    ///  A node of our binary tree
    /// </summary>
    internal class TreeNode
    {
        private Item _item = null;
        private TreeNode _left = null;
        private TreeNode _right = null;

        public Item Item
        {
            get { return _item; }
        }

        public TreeNode Left
        {
            get { return _left; }
        }

        public TreeNode Right
        {
            get { return _right; }
        }

        public TreeNode(Item item, TreeNode left, TreeNode right)
        {
            _item = item;
            _left = left;
            _right = right;
        }
    }
}

File: Code/Iterator.Java/src/datastructure/TreeNode.java
package datastructure;

import model.Item;

/**
 * A node of our binary tree
 */
public class TreeNode {
	
	private Item _item = null;
	private TreeNode _left = null;
	private TreeNode _right = null;

	public Item getItem() {
		return _item;
	}

	public TreeNode getLeft() {
		return _left;
	}
	
	public TreeNode getRight() {
		return _right;
	}
	
	public TreeNode(Item item, TreeNode left, TreeNode right) {
		
		this._item = item;
		this._left = left;
		this._right = right;

	}

}
